File: api/pacientes/src/main/java/com/stp/pacientes/model/Paciente.java
package com.stp.pacientes.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class Paciente {
    @Id
    @Column(length = 11, nullable = false, unique = true)
    private String cpf;
    private String nome;
    private String telefone;
    private String email;
    @Embedded
    private Endereco endereco;
    @OneToOne(cascade = CascadeType.ALL)
    private Ficha ficha;
}

File: app/app/src/main/java/com/stp/pacientes/api/Paciente.kt
package com.stp.pacientes.api

data class Endereco(
    val numero: String,
    val cep: String,
    val rua: String,
    val bairro: String,
    val cidade: String,
    val estado: String
)

data class Ficha(
    val medicamentos: String,
    val historico: String,
    val alergias: String
)

data class Paciente(
    val cpf: String,
    val nome: String,
    val telefone: String,
    val email: String,
    val endereco: Endereco,
    val ficha: Ficha
)
